tests/sources/test_redis.py

```diff
@@ -213,3 +218,105 @@ async def test_get_db_records():
         source.client.get_paginated_key = AsyncIterator(["0"])
         async for record in source.get_db_records(db=0):
             assert record == DOCUMENT[0]
+
+
+@pytest.mark.parametrize(
+    "filtering",
+    [
+        Filter(
+            {
+                ADVANCED_SNIPPET: {
+                    "value": [
+                        {"database": 0},
+                    ]
+                }
+            }
+        ),
+    ],
+)
+@pytest.mark.asyncio
+@freeze_time("2023-01-24T04:07:19+00:00")
+async def test_get_docs_with_sync_rules(filtering):
+    async with create_redis_source() as source:
+        source.client.database = ["*"]
+        source.get_db_records = AsyncIterator(items=DOCUMENT)
+        async for (doc, _) in source.get_docs(filtering):
+            assert doc in DOCUMENT
+
+
+@pytest.mark.parametrize(
+    "advanced_rules, expected_validation_result",
+    [
+        (
+            # valid: empty array should be valid
+            [],
+            SyncRuleValidationResult.valid_result(
+                SyncRuleValidationResult.ADVANCED_RULES
+            ),
+        ),
+        (
+            # valid: empty object should also be valid -> default value in Kibana
+            {},
+            SyncRuleValidationResult.valid_result(
+                SyncRuleValidationResult.ADVANCED_RULES
+            ),
+        ),
+        (
+            # valid: one custom pattern
+            [{"database": 0, "key_pattern": "*"}],
+            SyncRuleValidationResult.valid_result(
+                SyncRuleValidationResult.ADVANCED_RULES
+            ),
+        ),
+        (
+            # valid: two custom patterns
+            [
+                {"database": 0, "key_pattern": "test*"},
+                {"database": 1, "_type": "string"},
+            ],
+            SyncRuleValidationResult.valid_result(
+                SyncRuleValidationResult.ADVANCED_RULES
+            ),
+        ),
+        (
+            # invalid: database number
+            [{"database": -1}],
+            SyncRuleValidationResult(
+                SyncRuleValidationResult.ADVANCED_RULES,
+                is_valid=False,
+                validation_message=ANY,
+            ),
+        ),
+        (
+            # invalid: array of arrays -> wrong type
+            {"database": ["a/b/c", ""]},
+            SyncRuleValidationResult(
+                SyncRuleValidationResult.ADVANCED_RULES,
+                is_valid=False,
+                validation_message=ANY,
+            ),
+        ),
```

Comment:
Add a test case with only `database` configured

Reply:
Last second parameter has only database configured

Reply:
Did you mean this?
```
{"database": ["a/b/c", ""]}
```
I think this will be rejected by 
```
"database": {"type": "integer", "minLength": 1, "minimum": 0}
```

But it won't test 
```
"anyOf": [
    {"required": ["key_pattern"]},
    {"required": ["_type"]},
]
```

Reply:
added